File: app/src/main/java/knf/kuma/directory/DirectoryViewModel.java
package knf.kuma.directory;

import android.content.Context;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;
import androidx.paging.PagedList;
import knf.kuma.pojos.AnimeObject;
import knf.kuma.retrofit.Repository;

public class DirectoryViewModel extends ViewModel {
    private Repository repository=new Repository();

    public LiveData<PagedList<AnimeObject>> getAnimes(Context context) {
        return repository.getAnimeDir(context);
    }

    public LiveData<PagedList<AnimeObject>> getOvas(Context context) {
        return repository.getOvaDir(context);
    }

    public LiveData<PagedList<AnimeObject>> getMovies(Context context) {
        return repository.getMovieDir(context);
    }
}

File: app/src/main/java/knf/kuma/directory/DirectoryViewModel.kt
package knf.kuma.directory

import androidx.lifecycle.ViewModel
import androidx.paging.PagingData
import knf.kuma.retrofit.Repository
import kotlinx.coroutines.flow.Flow

class DirectoryViewModel : ViewModel() {
    private val repository = Repository()

    fun getAnimes(): Flow<PagingData<DirObject>> {
        return repository.getAnimeDir()
    }

    fun getOvas(): Flow<PagingData<DirObject>> {
        return repository.getOvaDir()
    }

    fun getMovies(): Flow<PagingData<DirObject>> {
        return repository.getMovieDir()
    }
}
